internal/utils/utils.go

```diff
@@ -25,11 +22,6 @@ func CheckVersion(version string) error {
 	return nil
 }
 
-func ParseCiliumVersion(version string) (semver.Version, error) {
```

Comment:
Should we consider keeping this function and just modifying it to use `semver.ParseTolerant` instead of `versioncheck.Version`? Should we fix version check upstream instead? 

Also, I think we may still want the `strings.TrimPrefix`.

Reply:
ok yeah i can do that.

> Also, I think we may still want the strings.TrimPrefix.

`semver.ParseTolerant` already trims the `v` prefix ✅